src/diffusers/loaders.py

```diff
@@ -11,3326 +11,84 @@
-
-    # overwrite PyTorch's `state_dict` to be sure that only the 'regular_linear_layer' weights are saved
-    # when saving the whole text encoder model and when LoRA is unloaded or fused
-    def state_dict(self, *args, destination=None, prefix="", keep_vars=False):
-        if self.lora_linear_layer is None:
-            return self.regular_linear_layer.state_dict(
-                *args, destination=destination, prefix=prefix, keep_vars=keep_vars
-            )
-
-        return super().state_dict(*args, destination=destination, prefix=prefix, keep_vars=keep_vars)
-
-    def _fuse_lora(self, lora_scale=1.0, safe_fusing=False):
-        if self.lora_linear_layer is None:
-            return
-
-        dtype, device = self.regular_linear_layer.weight.data.dtype, self.regular_linear_layer.weight.data.device
-
-        w_orig = self.regular_linear_layer.weight.data.float()
-        w_up = self.lora_linear_layer.up.weight.data.float()
-        w_down = self.lora_linear_layer.down.weight.data.float()
-
-        if self.lora_linear_layer.network_alpha is not None:
-            w_up = w_up * self.lora_linear_layer.network_alpha / self.lora_linear_layer.rank
-
-        fused_weight = w_orig + (lora_scale * torch.bmm(w_up[None, :], w_down[None, :])[0])
-
-        if safe_fusing and torch.isnan(fused_weight).any().item():
-            raise ValueError(
-                "This LoRA weight seems to be broken. "
-                f"Encountered NaN values when trying to fuse LoRA weights for {self}."
-                "LoRA weights will not be fused."
-            )
-
-        self.regular_linear_layer.weight.data = fused_weight.to(device=device, dtype=dtype)
-
-        # we can drop the lora layer now
-        self.lora_linear_layer = None
-
-        # offload the up and down matrices to CPU to not blow the memory
-        self.w_up = w_up.cpu()
-        self.w_down = w_down.cpu()
-        self.lora_scale = lora_scale
-
-    def _unfuse_lora(self):
-        if not (getattr(self, "w_up", None) is not None and getattr(self, "w_down", None) is not None):
-            return
-
-        fused_weight = self.regular_linear_layer.weight.data
-        dtype, device = fused_weight.dtype, fused_weight.device
-
-        w_up = self.w_up.to(device=device).float()
-        w_down = self.w_down.to(device).float()
-
-        unfused_weight = fused_weight.float() - (self.lora_scale * torch.bmm(w_up[None, :], w_down[None, :])[0])
-        self.regular_linear_layer.weight.data = unfused_weight.to(device=device, dtype=dtype)
-
-        self.w_up = None
-        self.w_down = None
-
-    def forward(self, input):
-        if self.lora_scale is None:
-            self.lora_scale = 1.0
-        if self.lora_linear_layer is None:
-            return self.regular_linear_layer(input)
-        return self.regular_linear_layer(input) + (self.lora_scale * self.lora_linear_layer(input))
+from .loaders.lora import LoraLoaderMixin, StableDiffusionXLLoraLoaderMixin
```

Comment:
Let's fully delete this file. There should be no backward breaking change as:

```py
from diffusers.loaders import UNet2DConditionLoadersMixin
```

is still supported since you have created a `loaders/__init__.py` file. 
I think we can safely delete this file

Reply:
Done in ca011fd2.